### ISSUE TITLE
Investigate how to make this thing work with any debuggee

### ISSUE THREAD

**Issue by captainbrosset** [captainbrosset/domprofiler#10]
For now this will only work with a local tab because the recorder lives in a frame script that is inserted by the panel (see https://developer.mozilla.org/en-US/docs/The_message_manager).
If we want to have this work with any device/process/whatever, the recorder needs to be a protocol.js actor. I don't think there is yet a way to register new actors with the firefox devtools though.

**Comment by dmose** [comment]
It would be extra hawt to be able to use this on other XUL iframes/browser elements, like those use by Firefox Hello.  Perhaps somebody in #devtools has suggestions on new actor registration?

**Comment by captainbrosset** [comment]
Yeah that would be great.
@janodvarko I'm not exactly aware about the plan to let addon define and load debugger-server actors. Is this something we can do already? Any ongoing work being done?

**Comment by janodvarko** [comment]
@captainbrosset Yes, I am working on a new platform API that will allow dynamic registration (and unregistration) of actors coming from addons. 

See https://bugzilla.mozilla.org/show_bug.cgi?id=977443
(the patch is waiting for a review)

Honza

**Comment by captainbrosset** [comment]
Awesome, thanks for the link. I'll follow this issue closely. This is a must have API for addons.

**Comment by captainbrosset** [comment]
https://bugzilla.mozilla.org/show_bug.cgi?id=977443 just got fixed in nightly!! Time for some custom actor registration goodness!

**Comment by dmose** [comment]
Woot!

2014-11-20 15:14 GMT-06:00 Patrick Brosset notifications@github.com:

**Comment by janodvarko** [comment]
Yep, this is one of the basic stones for custom remote features.
Just to note that e10s support for the dynamic actor registration isn't yet ready.
Honza
